usr/src/uts/common/fs/zfs/sys/spa.h

```diff
@@ -326,7 +327,8 @@ typedef enum bp_embedded_type {
 typedef struct blkptr {
 	dva_t		blk_dva[SPA_DVAS_PER_BP]; /* Data Virtual Addresses */
 	uint64_t	blk_prop;	/* size, compression, type, etc	    */
-	uint64_t	blk_pad[2];	/* Extra space for the future	    */
+	uint32_t	blk_iv[3];	/* IV for block level encryption    */
+	uint32_t	blk_pad;	/* Extra space for the future	    */
```

Comment:
If DVA2 is a feasible place for crypto (which it is by definition since Oracle is already doing that), and that avoids burning the padding, I'd like to hear more about whether Nexenta's plans absolutely require the use of DVA2, or if maybe that can be relocated elsewhere such that the padding can be preserved and both crypto and Nexenta's feature can coexist. This should be decided based on what's best technically, not based on who called "shotgun" first.